google/resource_sql_database_instance.go

```diff
@@ -638,7 +647,17 @@ func resourceSqlDatabaseInstanceRead(d *schema.ResourceData, meta interface{}) e
 	}
 
 	_settingsList := d.Get("settings").([]interface{})
-	_settings := _settingsList[0].(map[string]interface{})
+	var _settings map[string]interface{}
+
+	// If we're importing the settings will be nil
+	if len(_settingsList) > 0 {
+		_settings = _settingsList[0].(map[string]interface{})
+	} else {
+		_settingsList = make([]interface{}, 1)
```

Comment:
It looks like when we use the  `_settingsList` slice down there, we are just overwriting it's contents with the `_settings` object. We can introduce a new slice to wrap `_settings` in there, and better semantically show that the two slices are not linked. That way, we can eliminate this line and break an implicit dependency between these parts of the function.